src/main/java/uk/gov/laa/ccms/caab/controller/notifications/ActionsAndNotificationsController.java

```diff
@@ -171,28 +184,172 @@ public String getNotification(
+      @ModelAttribute(USER_DETAILS) UserDetail user,
+      @PathVariable(value = "notification_id") String notificationId,
+      @PathVariable(value = "attachment_id") Integer attachmentId) {
+
+    notificationService.retrieveDraftNotificationAttachment(String.valueOf(attachmentId),
+        user.getLoginId(),
+        user.getUserType());
+
+    return "redirect:/notifications/%s/provide-documents-or-evidence".formatted(notificationId);
+
+  }
+
+  /**
+   * If the cover sheet does not exist in S3, retrieve it from EBS then upload to S3.
+   *
+   * @param user           current user details.
+   * @param notificationId the ID of the notification of which the attachment belongs to.
+   * @param attachmentId   the ID of the notification attachment to retrieve.
+   * @return the notification page.
+   */
+  @GetMapping("/notifications/{notification_id}/attachments/{attachment_id}/retrieveCoverSheet")
+  public String retrieveCoverSheet(
+      @ModelAttribute(USER_DETAILS) UserDetail user,
+      @PathVariable(value = "notification_id") String notificationId,
+      @PathVariable(value = "attachment_id") Integer attachmentId) {
+
+    notificationService.retrieveCoverSheet(String.valueOf(attachmentId),
+        user.getLoginId(),
+        user.getUserType());
+
+    return "redirect:/notifications/%s/provide-documents-or-evidence".formatted(notificationId);
+  }
+
+  /**
+   * Remove a draft notification attachment from TDS and S3.
+   *
+   * @param user           the currently logged-in user.
+   * @param notificationId the ID of the notification.
+   * @param attachmentId   the ID of the notification attachment to remove.
+   * @return the provide documents or evidence page.
+   */
+  @GetMapping("/notifications/{notification_id}/attachments/{attachment_id}/remove")
+  public String removeDraftNotificationAttachment(
+      @ModelAttribute(USER_DETAILS) UserDetail user,
+      @PathVariable(value = "notification_id") String notificationId,
+      @PathVariable(value = "attachment_id") Integer attachmentId) {
+
+    notificationService.removeDraftNotificationAttachment(notificationId, attachmentId,
+        user.getLoginId(), user.getUserId());
+    return "redirect:/notifications/%s/provide-documents-or-evidence".formatted(notificationId);
+  }
+
+  /**
+   * Display the edit notification attachment screen.
+   *
+   * @param user           the currently logged-in user.
+   * @param notificationId the ID of the notification.
+   * @param attachmentId   the ID of the notification attachment.
+   * @return the edit notification attachment screen.
+   */
+  @GetMapping("/notifications/{notification_id}/attachments/{attachment_id}/edit")
+  public String editDraftNotificationAttachment(
+      @ModelAttribute(USER_DETAILS) UserDetail user,
+      @PathVariable(value = "notification_id") String notificationId,
+      @PathVariable(value = "attachment_id") Integer attachmentId) {
+
+    // TODO: CCLS-1935 & CCLS-1937
```

Comment:
Is this todo still required?

Reply:
I'm doing those tickets in the current sprint so this will be removed in the next PR.